bin/dev-tools/create_shipments.py

```diff
@@ -69,184 +111,180 @@ class CreateShipments:
         "timestamp": datetime.now(timezone.utc).isoformat(),
     }
 
-    # Values that can be set via command line
-    attributes = {}
-
-    # Values that cannot be set via command line but dynamic within function
-    errors = []
-    shipments = []
-
-    def __init__(self):
-        self.active_users = {'user_1'}
-        args = parser.parse_args()
-        self.carrier = self._validate_user(args.carrier) if args.carrier else self.users['user_1']
-        self.shipper = self._validate_user(args.shipper) if args.shipper else self.users['user_1']
-        self.moderator = self._validate_user(args.moderator) if args.moderator else None
-        self.sequence_number = args.startnumber if args.startnumber else 0
-        self.total = int(args.total) if args.total else 10
-        self.verbose = args.verbose
-        self.device = args.device
-        self.chunk_size = int(args.partition) if args.partition else 10
-        self.add_tracking = args.add_tracking
+    # pylint: disable=too-many-arguments, too-many-locals
+    def __init__(self, carrier='user1@shipchain.io', shipper='user1@shipchain.io', moderator=None, sequence_number=0,
+                 total=10, loglevel='warning', device=False, partition=10, attributes=None, add_tracking=False,
+                 add_telemetry=False, profiles_url='http://localhost:9000', transmission_url='http://localhost:8000'):
+        self.attributes = {}
+        self.errors = []
+        self.shipments = []
+        self.carrier = self.users[carrier]
+        self.shipper = self.users[shipper]
+        self.moderator = self.users[moderator] if moderator else None
+        self.sequence_number = sequence_number
+        self.total = total
+        console = logging.StreamHandler()
+        console.setLevel(LogLevels[loglevel].value)
+        logger.addHandler(console)
+        logger.setLevel(LogLevels[loglevel].value)
+        self.device = device
+        self.chunk_size = partition
+        self.add_tracking = add_tracking
+        self.add_telemetry = add_telemetry
+        self.profiles_url = profiles_url
+        self.transmission_url = transmission_url
+        if attributes:
+            self._validate_attributes(attributes)
+
+    def handle_args(self, command_args):
```

Comment:
this can be done very easily so making the change now. I figured that it made sense to put all the logic for setting the variables in the `__init__` method since they were all required for the `ShipmentCreator` to run. I can move the `__init__` logic into another function designed to parse the variables given, but it does run into the potential issue where someone could create an instance of the `ShipmentCreator` object and then not passing in the proper values.

Reply:
Do you think it would be better to move the logic outside of the `__init__` into a new function `set_shipment_fields` and have the fields be passed in through there instead?